inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_binary_elementwise.cpp

```diff
@@ -48,7 +48,25 @@ void processBinaryEltwise(std::shared_ptr<ngraph::Node> eltwise, size_t lhsIndex
         broadcastInput = std::make_shared<ngraph::opset3::Concat>(ngraph::OutputVector{broadcastConst, broadcastInput}, 0);
     }
 
-    const auto shape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& lhsInputShape = lhsInput.get_partial_shape();
+    const auto& rhsInputShape = rhsInput.get_partial_shape();
+
+    VPU_THROW_UNLESS(lhsInputShape.is_static() && lhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects lhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), lhsInputShape);
+
+    VPU_THROW_UNLESS(rhsInputShape.is_static() && rhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects rhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), rhsInputShape);
+
+    const auto& lhsMin = std::make_shared<ngraph::opset5::ReduceMin>(lhsInput, ngraph::opset5::Constant::create(shapeElementType, {}, {0}));
+    const auto& rhsMin = std::make_shared<ngraph::opset5::ReduceMin>(rhsInput, ngraph::opset5::Constant::create(shapeElementType, {}, {0}));
+    const auto maxRank = std::max(ngraph::shape_size(lhsInputShape.to_shape()), ngraph::shape_size(lhsInputShape.to_shape()));
+    const auto& zeros = std::vector<std::int64_t>(maxRank, 0);
+    const auto& shape = std::make_shared<ngraph::opset5::Select>(
+        std::make_shared<ngraph::opset5::Convert>(std::make_shared<ngraph::opset5::Minimum>(lhsMin, rhsMin), ngraph::element::boolean),
+        std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput),
+        ngraph::opset5::Constant::create(shapeElementType, {static_cast<std::size_t>(maxRank)}, zeros));
```

Comment:
@slyalin @lazarevevgeny 

What should we do if we're trying to execute Add on inputs with shapes {0} and {1}?

Reply:
Discussed offline. {1} is broadcasted to {0} so solution meets specification criteria.